dbms/src/Interpreters/Join.h

```diff
@@ -54,25 +64,38 @@ struct WithFlags<T, false> : T
 
     void setUsed() const {}
     bool getUsed() const { return true; }
+    bool setUsedOnce() const { return true; }
 };
 
-using MappedAny =       WithFlags<RowRef, false>;
-using MappedAll =       WithFlags<RowRefList, false>;
-using MappedAnyFull =   WithFlags<RowRef, true>;
-using MappedAllFull =   WithFlags<RowRefList, true>;
-using MappedAsof =      WithFlags<AsofRowRefs, false>;
+using MappedOne =        WithFlags<RowRef, false>;
+using MappedAll =        WithFlags<RowRefList, false>;
+using MappedOneFlagged = WithFlags<RowRef, true>;
+using MappedAllFlagged = WithFlags<RowRefList, true>;
+using MappedAsof =       WithFlags<AsofRowRefs, false>;
 
 }
 
 /** Data structure for implementation of JOIN.
   * It is just a hash table: keys -> rows of joined ("right") table.
   * Additionally, CROSS JOIN is supported: instead of hash table, it use just set of blocks without keys.
   *
-  * JOIN-s could be of nine types: ANY/ALL × LEFT/INNER/RIGHT/FULL, and also CROSS.
+  * JOIN-s could be of these types:
```

Comment:
Please-please add some examples of small joined tables in comments or docs (maybe later).
And also information about equivalent joins.

Reply:
We have to make it in docs.
If you mean old ANY INNER vs SEMI LEFT JOIN equivalence. It's only with any_join_distinct_right_table_keys setted to true. And it sould be removed in future.

Other equivalences such as ```t INNER JOIN u``` <=> ```u INNER JOIN t``` and ```t LEFT JOIN u``` <=> ```u RIGHT JOIN t``` are expected.